File: DP/N_step.cpp
#include <iostream>
using namespace std;

long long F[91];

int main(){
    F[1] = 1;
    F[2] = 2;

    for(int i=3;i<=90;i++){
        F[i] = F[i-1] + F[i-2];
    }
    int n;
    cout << "please input n:" <<endl;
    while(cin >> n){
        if (n>=90)
            break;
        cout<< F[n] << endl;
        cout << "please input n:" <<endl;
    }

    return 0;
}

File: DP/N_step.py

F = [0, 1, 2]


def N_step(n):
    if n < 0:
        return -1
    for i in range(3, 90):
        x = F[i - 1] + F[i - 2]
        F.insert(i, x)
    return F[n]


if __name__ == '__main__':
    n = int(input('please input n:'))
    while(n < 90):
        print(N_step(n))
        n = int(input('please input n:'))
